### PR TITLE
ludusavi: init at 0.15.0

### PR DESCRIPTION
**Pull request by pasqui23**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/backup/ludusavi/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib, rustPlatform, fetchCrate, cmake, pkg-config, freetype, expat,libX11 ,xorg,fontconfig}:
+rustPlatform.buildRustPackage rec {
```
Comment (azahi) [inline pkgs/applications/backup/ludusavi/default.nix:2]:
```suggestion
{ lib
, rustPlatform
, fetchCrate
, cmake
, expat
, fontconfig
, freetype
, libX11
, libXcursor
, libXi
, libXrandr
, libxcb
, pkg-config
}:

rustPlatform.buildRustPackage rec {
```

File: pkgs/applications/backup/ludusavi/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  pname = "ludusavi";
+  version = "0.12.1";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-ipZW3p+/Pqam4sAYLXMKMibjEH0pMCDfOFAZaCqes3k=";
```
Comment (azahi) [inline pkgs/applications/backup/ludusavi/default.nix:8]:
```suggestion
    hash = "sha256-ipZW3p+/Pqam4sAYLXMKMibjEH0pMCDfOFAZaCqes3k=";
```

File: pkgs/applications/backup/ludusavi/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  };
+
+  cargoHash = "sha256-CPbGCcPtJt0ZqKNLv8WRRucMCJ8a5xXk5X5FLoqbmNA=";
+  cargoDepsName = pname;
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [ freetype expat libX11 xorg.libxcb xorg.libXcursor xorg.libXrandr xorg.libXi fontconfig];
```
Comment (azahi) [inline pkgs/applications/backup/ludusavi/default.nix:14]:
```suggestion
  buildInputs = [
    expat
    fontconfig
    freetype
    libX11
    libXcursor
    libXi
    libXrandr
    libxcb
 ];
```

File: pkgs/applications/backup/ludusavi/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  meta = with lib; {
+    description = "Backup tool for PC game saves";
+    homepage = "https://github.com/mtkennerly/ludusavi";
```
Comment (azahi) [inline pkgs/applications/backup/ludusavi/default.nix:18]:
```suggestion
    homepage = "https://github.com/mtkennerly/ludusavi";
    changelog = "https://github.com/mtkennerly/ludusavi/blob/v${version}/CHANGELOG.md"
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/applications/backup/ludusavi/default.nix (hunk 2 above)
Comment (pasqui23) [reply pkgs/applications/backup/ludusavi/default.nix:8]: `fetchCrate` does not have a `hash` input

**Comment by onny** [comment]
Please squash commits :)

**Comment by SuperSamus** [comment]
GUI doesn't work (simply typing `ludusavi` should open the GUI).
[I made a branch that fixes the issue](https://github.com/SuperSamus/nixpkgs/blob/ludusavi/pkgs/applications/backup/ludusavi/default.nix).
Not sure if some of the stuff should be _only_ in either `buldInputs` or `libPath` (instead of both).
File picker doesn't work (not the biggest deal), don't know if it's related to https://github.com/Seeker14491/opener/issues/4.

**Comment by SuperSamus** [comment]
Found the cause for the not working file picker, and fixed it. (Again, on [my branch](https://github.com/SuperSamus/nixpkgs/blob/ludusavi/pkgs/applications/backup/ludusavi/default.nix)).
(BTW, don't forget to update the PR title)

**Comment by SuperSamus** [comment]
I think that you need to squash the commits, and to rename the PR.

**Comment by SuperSamus** [comment]
You probably need to rebase, you are more than 17000 commits behind upstream master.
(And you need to rename the tile once again)

**Review by SuperSamus — APPROVED** [review]

**Pull request merged**